app/src/actions/queryActions.test.js

```diff
@@ -18,103 +18,304 @@ describe('query actions', () => {
+        )
+      );
 
-    const store = mockStore({ query: {} });
+    const store = mockStore({ query: { metadata: {} } });
 
-    store.dispatch(actions.fetchQuery('doesntMatter')).then(() => {
-      //now we're expecting the action to have been triggered with particular
-      //checking the length because it's not otherwise so easy to compare raw
-      //network response vs. parsed data, will test that transformation logic
-      //in reducer tests
-      expect(store.getActions()[0].results).toHaveLength(MOCK_RESP_DUPE_LEN)
-    });
+    const fn = actions.fetchQuery('doesntMatter');
+    expect(fn).toBeInstanceOf(Function);
+    fn(store.dispatch, store.getState);
+
+    //now we're expecting the action to have been triggered with particular
+    //checking the length because it's not otherwise so easy to compare raw
+    //network response vs. parsed data, will test that transformation logic
+    //in reducer tests
+    setTimeout(() => {
+      expect(store.getActions()[1].results).toHaveLength(MOCK_RESP_DUPE_LEN);
+      done();
+    }, 3);
+  });
 
+  it('should be able to handle fetch query with empty result', done => {
     //re-mock fetch with diff mock value, an empty response
-    window.fetch = jest.fn().mockImplementation(() =>
-      Promise.resolve(new Response(JSON.stringify(MOCK_RESP_EMPTY), {status:200})));
-    return store.dispatch(actions.fetchQuery('doesntMatter')).then(() => {
-      //each fetchQuery will trigger 2 actions. first is requestQuery and second is receiveQuery which has the results
-      //now we're looking for receiveQuery (the 3nd as there was another fetchQuery earlier) action to have been triggered
-      expect(store.getActions()[3].results).toHaveLength(0)
+    window.fetch = jest
+      .fn()
+      .mockImplementation(() =>
+        Promise.resolve(
+          new Response(JSON.stringify(MOCK_RESP_EMPTY), { status: 200 })
+        )
+      );
+
+    const store = mockStore({ query: { metadata: {} } });
+    const fn = actions.fetchQuery('doesntMatter');
+    expect(fn).toBeInstanceOf(Function);
+    fn(store.dispatch, store.getState);
+
+    //each fetchQuery will trigger 2 actions. first is requestQuery and second is receiveQuery which has the results
+    //now we're looking for receiveQuery (the 3nd as there was another fetchQuery earlier) action to have been triggered
+    setTimeout(() => {
+      expect(store.getActions()[1].results).toHaveLength(0);
+      done();
+    }, 3);
+  });
+
+  it('should handle fetch metadata', done => {
+    const store = mockStore({ query: { metadata: {} } });
+
+    window.fetch = jest
+      .fn()
+      .mockImplementation(() =>
+        Promise.resolve(
+          new Response(JSON.stringify(APP_METADATA), { status: 200 })
+        )
+      );
+
+    const fn = actions.fetchMetadata('application');
+    expect(fn).toBeInstanceOf(Function);
+    fn(store.dispatch, store.getState);
+
+    setTimeout(() => {
```

Comment:
this is fine since you're correctly using the `done` method but I think there is a different and more elegant way to test actions' impacts on the store synchronously that I read about recently. let me find that back and will share